### PR TITLE
Test new BCL measures

### PR DESCRIPTION
**Pull request by joseph-robertson**
## Pull Request Description

Test new measures using `national_baseline.yml`:
* AddThermalComfortModelTypes
* PeakPeriodSchedulesShift

## Checklist

Not all may apply:

- [ ] Tests (and test files) have been updated
- [ ] Documentation has been updated
- [ ] Changelog has been updated
- [ ] `openstudio tasks.rb update_measures` has been run
- [ ] No unexpected regression test changes on CI (checked comparison artifacts)

### CODE REVIEW

File: measures/GEBAppliancesPeakPeriodShift/measure.rb [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+    arg.setDisplayName('Schedules: Peak Period Delay')
+    arg.setUnits('hr')
+    arg.setDescription('The number of hours after peak period end.')
+    arg.setDefaultValue(0)
+    args << arg
+
+    arg = OpenStudio::Measure::OSArgument.makeStringArgument('schedules_peak_period_column_names', false)
+    arg.setDisplayName('Schedules: Peak Period Schedule File Column Names')
+    arg.setDescription('Comma-separated list of schedule file column names to shift during the peak period.')
+    args << arg
```
Comment (shorowit) [inline measures/GEBAppliancesPeakPeriodShift/measure.rb:46]: Should really clarify that this is specific to ScheduleFile objects.

Do you plan to support another argument similar to this where someone can enter a list of ScheduleRuleset names to shift? That is the most common schedule class used in OpenStudio after all, not that many users have schedules using ScheduleFile.

File: measures/GEBAppliancesPeakPeriodShift/measure.rb [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+        values = values.map { |v| Float(v) }
+      rescue ArgumentError
+        fail "Schedule value must be numeric for column '#{col_name}'. [context: #{schedules_path}]"
+      end
+
+      @schedules[col_name] = values
+    end
+  end
+
+  def shift_schedules(runner, schedule_file_column_names_enabled, begin_hour, end_hour, delay, total_days_in_year, sim_start_day, steps_in_day)
```
Comment (shorowit) [inline measures/GEBAppliancesPeakPeriodShift/measure.rb:144]: Need error-checking if schedule names are not found.

File: measures/GEBAppliancesPeakPeriodShift/measure.rb [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+# http://nrel.github.io/OpenStudio-user-documentation/reference/measure_writing_guide/
+
+require 'csv'
+
+# start the measure
+class GEBAppliancesPeakPeriodShift < OpenStudio::Measure::ModelMeasure
+  # human readable name
+  def name
+    # Measure name should be the title case of the class name.
+    return 'GEBAppliancesPeakPeriodShift'
```
Comment (shorowit) [inline measures/GEBAppliancesPeakPeriodShift/measure.rb:13]: Does this really only apply to appliances?

File: measures/GEBAppliancesPeakPeriodShift/measure.rb [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+    total_days_in_year = Schedules.NumDaysInYear(calendar_year)
+    sim_start_day = DateTime.new(calendar_year, 1, 1)
+
+    # get steps
+    ts = model.getTimestep
+    ts_per_hour = ts.numberOfTimestepsPerHour
+    steps_in_day = ts_per_hour * 24
+
+    model.getExternalFiles.each do |external_file|
+      external_file_path = external_file.filePath.to_s
```
Comment (shorowit) [inline measures/GEBAppliancesPeakPeriodShift/measure.rb:108]: Should probably check the schedule type. You wouldn't want to apply a shift to, e.g., a temperature schedule.

File: measures/GEBAppliancesPeakPeriodShift/measure.rb (hunk 1 above)
Comment (joseph-robertson) [reply measures/GEBAppliancesPeakPeriodShift/measure.rb:46]: Definitely really early for a review. I'm more testing the overall mechanics here. But yes, my plan is to get the basic functionality in for supporting residential/resstock needs, and then move onto some of the items you are highlighting in these comments...

File: measures/GEBAppliancesPeakPeriodShift/measure.rb (hunk 4 above)
Comment (joseph-robertson) [reply measures/GEBAppliancesPeakPeriodShift/measure.rb:108]: The burden is on the user to not provide the name of a temperature schedule.

File: measures/GEBAppliancesPeakPeriodShift/measure.rb (hunk 1 above)
Comment (shorowit) [reply measures/GEBAppliancesPeakPeriodShift/measure.rb:46]: Got it. Yep, I wasn't really reviewing the code, mostly just putting down some thoughts prompted by this PR.

File: measures/GEBAppliancesPeakPeriodShift/measure.rb (hunk 3 above)
Comment (joseph-robertson) [reply measures/GEBAppliancesPeakPeriodShift/measure.rb:13]: Good point. Something like "SchedulesPeakPeriodShift" probably more appropriate.

**Pull request closed without merging**